msystems/services.py

```diff
@@ -84,6 +82,22 @@ def _update_user_legal_entities(self, user: User, user_data: dict) -> None:
         self._delete_old_user_policyholders(user, policyholders)
         self._add_new_user_policyholders(user, policyholders)
 
+    def _update_user_roles(self, user, user_data):
+        msystem_roles = user_data.get('Role')
+        current_user_roles = Role.objects.filter(user_roles__user=user.i_user).values_list('is_system', flat=True)
+        current_user_roles_list = list(current_user_roles)
+        incoming_imis_role_ids = [self._parse_msystem_role_to_imis_role_id(msystem_role_id) for msystem_role_id in
+                                  msystem_roles]
```

Comment:
As above, don't use is_system as identifying field for the role.
Please check validity_to in user_roles table to filter out revoked roles.

Reply:
done